### ISSUE TITLE
Can't encode default theme data.

### ISSUE THREAD

**Issue by bimsina** [peiffer-innovations/json_theme#18]
```
 var encodedTheme = ThemeEncoder.encodeThemeData(ThemeData.light());
 print(encodedTheme);
```

The above code produces the following exception:

```
'package:json_theme/src/codec/theme_encoder.dart': Failed assertion: line 3546 pos 12: 'value == null ||
        value is CircleBorder ||
        value is ContinuousRectangleBorder ||
        value is RoundedRectangleBorder': is not true.
```

**Comment by jpeiffer** [comment]
Just to confirm, can you add in the information from `flutter doctor`?

**Comment by bimsina** [comment]
```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 1.22.2, on Mac OS X 10.15.7 19H2, locale en-NP)
 
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
[✓] Xcode - develop for iOS and macOS (Xcode 12.1)
[✓] Android Studio (version 4.1)
[✓] VS Code (version 1.50.1)
[✓] Connected device (1 available)

• No issues found!
```

Also, it isn't just with the default themes, the following code also throws the same exception

```
Map<String, dynamic> themeJson = {};
var theme = ThemeDecoder.decodeThemeData(themeJson);
print(theme);
var encodedTheme = ThemeEncoder.encodeThemeData(theme);
print(encodedTheme);
```
It successfully decodes the json to a theme, but throws exception on encoding the same theme.

```
flutter: ThemeData#e9050

════════ Exception caught by gesture ═══════════════════════════════════════════
The following assertion was thrown while handling a gesture:
'package:json_theme/src/codec/theme_encoder.dart': Failed assertion: line 3546 pos 12: 'value == null ||
        value is CircleBorder ||
        value is ContinuousRectangleBorder ||
        value is RoundedRectangleBorder': is not true.
```

**Comment by jpeiffer** [comment]
Thanks for the info.  I just published a fix with [1.4.2](https://pub.dev/packages/json_theme).  Closing this, please feel free to add a new issue if you encounter any more problems.
